aws/resource_aws_cloudfront_public_key_test.go

```diff
@@ -44,16 +72,25 @@ func TestAccAWSCloudFrontPublicKey_namePrefix(t *testing.T) {
 			{
 				Config: testAccAWSCloudFrontPublicKeyConfig_namePrefix(),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckCloudFrontPublicKeyExistence("aws_cloudfront_public_key.example"),
+					testAccCheckCloudFrontPublicKeyExistence(resourceName),
 					resource.TestMatchResourceAttr("aws_cloudfront_public_key.example", "name", startsWithPrefix),
 				),
 			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"name_prefix",
+				},
```

Comment:
Not for this pull request, but we can support `name_prefix` import by adjusting the implementation to the `naming` package: https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/contribution-checklists.md#adding-resource-name-generation-support 👍